projects/js-packages/partner-coupon/components/redeem-partner-coupon-post-connection/index.jsx

```diff
@@ -0,0 +1,169 @@
+/**
+ * External dependencies
+ */
+import React, { useCallback, useState } from 'react';
+import PropTypes from 'prop-types';
+import { ActionButton, JetpackLogo } from '@automattic/jetpack-components';
+import { __, sprintf } from '@wordpress/i18n';
+import cookie from 'cookie';
+
+/**
+ * Internal dependencies
+ */
+import { usePartnerCouponRedemption } from '../../hooks';
+
+/**
+ * Import styles
+ */
+import './style.scss';
+
+export const DISMISS_COOKIE_NAME = 'jp-redeem-partner-coupon-dismissed';
+export const DISMISS_MAX_COOKIE_AGE = 24 * 60 * 60; // 1 day
+
+/**
+ * Is partner coupon redeem CTA dismissed?
+ *
+ * @returns {boolean} Is the redeem CTA dismissed?
+ */
+function isDismissed() {
+	const cookies = cookie.parse( document.cookie );
+	return !! cookies[ DISMISS_COOKIE_NAME ];
+}
+
+/**
+ * Dismiss partner coupon redeem CTA.
+ *
+ * @returns {void}
+ */
+function dismiss() {
+	document.cookie = cookie.serialize( DISMISS_COOKIE_NAME, true, {
```

Comment:
Should we worry about adding a `secure` attribute when the site is an HTTPS site?